src/system.js

```diff
@@ -141,22 +153,62 @@ module.exports = AFRAME.registerSystem('physics', {
   tick: function (t, dt) {
     if (!this.initialized || !dt) return;
 
+    const wrapperStartTime = Date.now();
+
     var i;
     var callbacks = this.callbacks;
 
     for (i = 0; i < this.callbacks.beforeStep.length; i++) {
       this.callbacks.beforeStep[i].beforeStep(t, dt);
     }
 
+    const engineStartTime = Date.now();
+
     this.driver.step(Math.min(dt / 1000, this.data.maxInterval));
     
+    const engineEndTime = Date.now();
+
+    if (this.trackPerf) {
+      this.cumulativeTimeEngine += engineEndTime;
+      this.cumulativeTimeEngine -= engineStartTime;
+      this.tickCounter++;
+    }
+    
     for (i = 0; i < callbacks.step.length; i++) {
       callbacks.step[i].step(t, dt);
     }
 
     for (i = 0; i < callbacks.afterStep.length; i++) {
       callbacks.afterStep[i].afterStep(t, dt);
     }
+
+    if (this.trackPerf) {
+      const wrapperEndTime = Date.now();
+      this.cumulativeTimeWrapper += wrapperEndTime;
+      this.cumulativeTimeWrapper -= wrapperStartTime;
+      this.cumulativeTimeWrapper -= engineEndTime;
+      this.cumulativeTimeWrapper += engineStartTime;
+
+      if (this.tickCounter === 100) {
+        if (this.statsToConsole) {
+          console.log(`Avg. physics tick duration (engine): ${this.cumulativeTimeEngine / 100} msecs`);
+          console.log(`Avg. physics tick duration (wrapper): ${this.cumulativeTimeWrapper / 100} msecs`);
+        }
+
+        if (this.statsToEvents) {
+          const engine = this.cumulativeTimeEngine / 100
+          const wrapper = this.cumulativeTimeWrapper / 100
+          const total = engine + wrapper
+          this.el.emit("physics-tick-timer", {engine: engine.toFixed(2),
+                                              wrapper: wrapper.toFixed(2), 
+                                              total: total.toFixed(2) })
```

Comment:
It creates a new object on each tick, so it will trigger the garbage collector more often so some measurement will be probably higher when it happens.
You should create the object in init
```
this.timerEventDetails = {}
```
and do this here:
```
this.timerEventDetails.engine = engine.toFixed(2)
this.timerEventDetails.wrapper = wrapper.toFixed(2)
this.timerEventDetails.total = total.toFixed(2)
this.el.emit("physics-tick-timer", this.timerEventDetails)
```

Reply:
Unless I'm mistaken, it creates a new object only on every 100th tick, which is why I hadn't worried too much about this.   But nevertheless, preallocating is more efficient, so I'll make that change.

Reply:
Ah yes you're right, every 100 ticks.